ted/import.hoon

```diff
@@ -0,0 +1,83 @@
+/-  spider, blog
+/+  *strandio
+=,  strand=strand:spider
+=,  strand-fail=strand-fail:libstrand:spider
+^-  thread:spider
+|=  arg=vase
+=/  m  (strand ,vase)
+^-  form:m
+;<  our=@p  bind:m  get-our
+;<  now=@da  bind:m  get-time
+=/  files-path=path  /(scot %p our)/blog/(scot %da now)/import
+=/  files
+  |-
+  =+  .^(=arch %cy files-path)
+  :: If there is no directory below this, read the file's contents
+  ?:  (test dir.arch ~)
+    =/  mark
+    ?+  (rear files-path)  !!
+      %md  wain
+      %html  @t
+      %css  @t
+      %noun  noun
+    ==
+    =/  content  .^(mark %cx files-path)
+    :-  [~ content] :: mimic fil.arch (unit item)
+    (~(urn by dir.arch) |=([name=@ta ~] ^$(files-path (snoc files-path name))))
+  :-  fil.arch  :: ~ or [~ u=item]
+  (~(urn by dir.arch) |=([name=@ta ~] ^$(files-path (snoc files-path name))))
+
+=/  all-files  ~(tap of files)
+=/  files-import  :+
+  published=*(map path [html=@t md=@t theme=@tas])
+  drafts=*(map path md=@t)
+  themes=*(map @tas css=@t)
+=.  files-import
```

Comment:
do a `=/  files-import` with a `^-` below it for the type. Or define the type at the top and do a `=/  files-import=the-type-specified-at-the-top`

Reply:
Wasn't exactly sure how to write this - we can talk about it.